### PR TITLE
Configurable cluster_idle_timeout_ms to bound upstream keepalives

### PR DESCRIPTION
**Pull request by esmet**
## Description
Add cluster_idle_timeout_ms to the ambassador module and Mapping schema, so that operators can set upstream keepalive timeouts.

## Related Issues
https://github.com/datawire/ambassador/issues/1542

## Testing
We use this for our use case. It ensures that Envoy times out idle connections before the typical upstream, to avoid the http1 race condition of missing an upstream close event at the moment a connection is chosen for proxying.

## Todos
- [ ] Tests
- [ ] Documentation

This needs some docs and probably a unit test.

Open question: should we allow this to be overriden per Mapping? Or does that make merging Mappings into a cluster more complicated?

### CODE REVIEW

File: ambassador/ambassador/envoy/v2/v2cluster.py [hunk 1]
```diff
@@ -35,11 +35,16 @@ def __init__(self, config: 'V2Config', cluster: IRCluster) -> None:
             else:
                 dns_lookup_family = 'V6_ONLY'
 
+        cluster_idle_timeout_ms = cluster.ir.ambassador_module.get('cluster_idle_timeout_ms', 30000)
```
Comment (esmet) [inline ambassador/ambassador/envoy/v2/v2cluster.py]: We might actually want a 60s default, which is what nginx does (http://nginx.org/en/docs/http/ngx_http_upstream_module.html#keepalive_timeout)

**Comment by kflynn** [comment]
I think that this should be configurable per-`Mapping`. It should be pretty apparent where you'll need to add code to split clusters -- if not, sing out!

**Comment by kflynn** [comment]
Taking this one too; I'm hoping to support timeouts for testing shortly.

**Comment by kflynn** [comment]
Whoops, jumped to gun there. @esmet, any joy on per-`Mapping` config here? 🙂

**Comment by esmet** [comment]
> Whoops, jumped the gun there. @esmet, any joy on per-`Mapping` config here? 🙂

I've been slacking on this. I'll try it today.

**Pull request merged**